<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Demo</class>
 <widget class="QMainWindow" name="Demo">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1699</width>
    <height>827</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>261</width>
      <height>121</height>
     </rect>
    </property>
    <property name="title">
     <string>Create Maze</string>
    </property>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>181</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>The width (Max 50)</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="widthEntry">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>20</y>
       <width>61</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>181</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>The height (Max 50)</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="heightEntry">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>40</y>
       <width>61</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="createMaze">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>80</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Create Maze</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>130</y>
      <width>261</width>
      <height>341</height>
     </rect>
    </property>
    <property name="title">
     <string>Algorithm</string>
    </property>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>121</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>The source coordinate:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>50</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>X(default X = 0)</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>70</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Y(default Y = 0)</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>90</y>
       <width>121</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>The destintion coordinate:</string>
     </property>
    </widget>
    <widget class="QLabel" name="labelEndX">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>110</y>
       <width>141</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>X(default X = width - 1)</string>
     </property>
    </widget>
    <widget class="QLabel" name="labelEndY">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>130</y>
       <width>141</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Y(default Y = height - 1)</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="startX">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>50</y>
       <width>81</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="startY">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>70</y>
       <width>81</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="endX">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>110</y>
       <width>81</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="endY">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>130</y>
       <width>81</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>220</y>
       <width>121</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Heuristic function</string>
     </property>
    </widget>
    <widget class="QPushButton" name="heuristicManhattan">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>290</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Start</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>290</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Heuristic (Manhattan)</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>250</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Heuristic (Euclid)</string>
     </property>
    </widget>
    <widget class="QPushButton" name="heuristicEuclid">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>250</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Start</string>
     </property>
    </widget>
    <widget class="QPushButton" name="displayCells">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>170</y>
       <width>181</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Display start and end cells</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>980</x>
      <y>10</y>
      <width>711</width>
      <height>741</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayoutGreedyBFS"/>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_3">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>10</y>
      <width>711</width>
      <height>741</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayoutAstar">
     <property name="topMargin">
      <number>0</number>
     </property>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1699</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
